sql/src/test/java/io/crate/integrationtests/UnionIntegrationTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.integrationtests;
+
+import io.crate.testing.TestingHelpers;
+import io.crate.testing.UseJdbc;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.junit.annotations.TestLogging;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.is;
+
+@ESIntegTestCase.ClusterScope(minNumDataNodes = 2)
+@UseJdbc
+public class UnionIntegrationTest extends SQLTransportIntegrationTest {
+
+    @Test
+    public void testUnionAll2SameTable() {
+        createColorsAndSizes();
+        execute("select * from sizes " +
+                "union all " +
+                "select * from sizes");
+        assertThat(Arrays.asList(response.rows()),
+            containsInAnyOrder(
+                new Object[]{"small"},
+                new Object[]{"large"},
+                new Object[]{"small"},
+                new Object[]{"large"}));
+    }
+
+    @Test
+    @TestLogging("io.crate.action.job:TRACE")
+    public void testUnionAll2Tables() {
+        createColorsAndSizes();
+        execute("select * from colors " +
+                "union all " +
+                "select * from sizes");
```

Comment:
think we can add a order here and use the more nice looking printedTable(). imho its enough to have 1 test without an order